lm_eval/models/bedrock.py

```diff
@@ -0,0 +1,224 @@
+import json
+from importlib.util import find_spec
+from typing import Any, List, Tuple
+
+from tqdm import tqdm
+
+from lm_eval import utils
+from lm_eval.api.model import LM
+from lm_eval.api.registry import register_model
+
+
+eval_logger = utils.eval_logger
+
+
+def bedrock_chat(
+    client,  # boto3.client("bedrock-runtime")
+    model: str,
+    prompt: str,
+    max_tokens: int,
+    temperature: float,
+    stop: List[str],
+    **kwargs: Any,
+) -> str:
+    """Wrapper function around the Bedrock chat completion API client with exponential back-off
+    in case of RateLimitError.
+
+    params:
+        client: boto3.Client
+            Bedrock runtime client
+        model: str
+            Bedrock model e.g. 'anthropic.claude-3-haiku-20240307-v1:0'
+        prompt: str
+            Prompt to feed to the model
+        max_tokens: int
+            Maximum number of tokens to sample from the model
+        temperature: float
+            Sampling temperature
+        stop: List[str]
+            List of stop sequences
+        kwargs: Any
+            Additional model_args to pass to the API client
+    """
+
+    if not find_spec("boto3"):
+        raise Exception(
+            "attempted to use 'bedrock' LM type, but package `boto3` is not installed. \
+please install boto3 via `pip install 'lm-eval[bedrock]'` or `pip install -e '.[bedrock]'`",
+        )
+
+    def _exception_callback(e: Exception, sleep_time: float) -> None:
+        eval_logger.warning(
+            f"RateLimitError occurred: {e.__cause__}\n Retrying in {sleep_time} seconds"
+        )
+
+    # @retry_on_specific_exceptions(
+    #     on_exceptions=[
+    #         client.ThrottlingException,
+    #         client.ModelTimeoutException,
+    #         client.InternalServerException,
+    #     ],
+    #     max_retries=None,  # retry forever, consider changing
+    #     on_exception_callback=_exception_callback,
+    # )
+    def messages():
+        # structured payload for request
+
+        formatted_messages = [
+            {"role": "user", "content": [{"type": "text", "text": prompt}]}
+        ]
+        body = {
+            "max_tokens": max_tokens,
+            "temperature": temperature,
+            "messages": formatted_messages,
+        }
+
+        # update body with params then encode
+        body.update(kwargs)
+        body_bytes = json.dumps(body).encode("utf-8")
+
+        response = client.invoke_model(
+            body=body_bytes,
+            contentType="application/json",
+            accept="application/json",
+            modelId=model,
+        )
+        response_body = json.loads(response.get("body").read())
+
+        return response_body["content"][0]["text"]
+
+    return messages()
+
+
+@register_model("bedrock")
+class BedrockChatLM(LM):
+    def __init__(
+        self,
+        batch_size: int = 1,
+        model: str = "anthropic.claude-3-haiku-20240307-v1:0",
```

Comment:
```suggestion
        model: str,
        batch_size: int = 1,
```

let's avoid having defaults for model engines!